lib/src/realm_class.dart

```diff
@@ -483,6 +483,40 @@ class Realm implements Finalizable {
       throw RealmError('Starting a write transaction on a frozen Realm is not allowed.');
     }
   }
+
+  /// Compacts a Realm file. A Realm file usually contains free/unused space.
+  ///
+  /// This method removes this free space and the file size is thereby reduced.
+  /// Objects within the Realm file are untouched.
+  /// Note: The file system should have free space for at least a copy of the Realm file. This method must not be called inside a transaction.
+  /// The Realm file is left untouched if any file operation fails.
+  static Future<bool> compact(Configuration config) {
+    late Configuration compactConfig;
+    if (config is LocalConfiguration) {
+      compactConfig = Configuration.local(config.schemaObjects.toList(),
+          schemaVersion: config.schemaVersion,
+          fifoFilesFallbackPath: config.fifoFilesFallbackPath,
+          path: config.path,
+          encryptionKey: config.encryptionKey,
+          disableFormatUpgrade: true,
+          isReadOnly: false);
```

Comment:
This will break the rule that read-only realms can not be compacted. You will always compact them. It must be `isReadOnly: config.isReadOnly`

Reply:
I agree. will fix